multi-int-store/src/test/java/com/connexta/multiintstore/MultiIntStoreIntegrationTest.java

```diff
@@ -100,17 +100,85 @@ public void testRetrieve() throws Exception {
   }
 
   @Test
-  public void testSearchService() {
+  public void testEmptySearchService() throws Exception {
+    mockMvc
+        .perform(MockMvcRequestBuilders.get("/search?q=searchKeyword"))
+        .andExpect(status().isOk())
+        .andExpect(content().json(new JSONArray().toString()));
+  }
+
+  @Test
+  public void testEmptySearchResults() throws Exception {
+    // given:
+    final String contents =
+        "All the color had been leached from Winterfell until only grey and white remained";
+    final String ingestId = "1234";
+
+    whenHttp(server)
+        .match(Condition.endsWithUri("/location/cst001"))
+        .then(
+            Action.contentType(MediaType.TEXT_PLAIN.toString()),
+            Action.stringContent(contents),
+            Action.status(HttpStatus.OK_200));
 
-    searchService.store(
-        new IndexedProductMetadata(
-            "1",
-            "All the color had been leached from Winterfell until only grey and white remained"));
+    mockMvc.perform(
+        MockMvcRequestBuilders.post("/store/" + ingestId)
+            .contentType("application/json")
+            .content(
+                TestUtil.createMetadataCallbackJson(
+                    "1234",
+                    "COMPLETE",
+                    "cst",
+                    MediaType.APPLICATION_JSON.toString(),
+                    256,
+                    "http://localhost:" + server.getPort() + "/location/cst001",
+                    "U",
+                    "ownerProducer")));
+
+    // verify:
+    mockMvc
+        .perform(MockMvcRequestBuilders.get("/search?q=thisKeywordDoesntMatchAnything"))
+        .andExpect(status().isOk())
+        .andExpect(content().json(new JSONArray().toString()));
+  }
 
-    List<IndexedProductMetadata> results = searchService.find("Winterfell");
+  @Test
+  public void testSearchResults() throws Exception {
+    // given:
+    final String contents =
+        "All the color had been leached from Winterfell until only grey and white remained";
+    final String ingestId = "1234";
 
-    searchService.delete("1");
+    whenHttp(server)
+        .match(Condition.endsWithUri("/location/cst001"))
+        .then(
+            Action.contentType(MediaType.TEXT_PLAIN.toString()),
+            Action.stringContent(contents),
+            Action.status(HttpStatus.OK_200));
 
-    results = searchService.find("Winterfell");
+    mockMvc.perform(
```

Comment:
At some point we will need to think about cleaning out our persistent stores like Solr and Cassandra between itests.

Reply:
See line 49

Reply:
I see that method on the interface. 
https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html